src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,221 @@
+import * as React from 'react'
+import getValidProps from '@helpscout/react-utils/dist/getValidProps'
+import propConnect from '../PropProvider/propConnect'
+import { classNames } from '../../utilities/classNames'
+import { namespaceComponent } from '../../utilities/component'
+import { noop } from '../../utilities/other'
+import { COMPONENT_KEY } from './Pagination.utils'
+import pluralize from '../../utilities/pluralize'
+
+import {
+  PaginationUI,
+  InformationUI,
+  NavigationUI,
+  RangeUI,
+  ButtonIconUI,
+} from './Pagination.css.js'
+import Text from '../Text'
+import Icon from '../Icon'
+import Button from '../Button'
+
+export interface Props {
+  activePage: number
+  className?: string
+  innerRef: (node: HTMLElement) => void
+  onChange: (nextPageNumber: number) => void
+  rangePerPage: number
+  separator?: string
+  showNavigation?: boolean
+  pluralizeSubject?: string
+  subject: string
+  totalItems: number
+}
+
+export class Pagination extends React.PureComponent<Props> {
+  static defaultProps = {
+    activePage: 1,
+    innerRef: noop,
+    onChange: noop,
+    rangePerPage: 50,
+    separator: 'of',
+    subject: '',
+    showNavigation: true,
+    totalItems: 0,
+  }
+
+  getNumberOfPages() {
+    const { rangePerPage, totalItems } = this.props
+    return Math.ceil(totalItems / rangePerPage)
+  }
+
+  getCurrentPage() {
+    const { activePage } = this.props
+    if (activePage < 1) {
+      return 1
+    }
+    return Math.min(this.getNumberOfPages(), Math.round(activePage))
+  }
+
+  getStartRange() {
+    const { rangePerPage } = this.props
+    const page = this.getCurrentPage()
+    return page * rangePerPage - rangePerPage + 1
```

Comment:
It might be good to come up with a default if there are no items (totalItems === 0), with the values: `totalItems: 0, rangePerPage: 50` currently you'll see:

`-49 - 0 of 0 Customer`
![](http://c.hlp.sc/462a556f46a3/Storybook%2002-15%2018-10.png)

Ideal should be perhaps?

`0 Customers`

Reply:
I updated how the component is rendered when totalItems ===0 in 6935457c